pkg/nfd-topology-updater/nfd-topology-updater.go

```diff
@@ -294,6 +295,7 @@ func (w *nfdTopologyUpdater) updateNodeResourceTopology(zoneInfo v1alpha2.ZoneLi
 		nrtNew := v1alpha2.NodeResourceTopology{
 			ObjectMeta: metav1.ObjectMeta{
 				Name:            w.nodeName,
+				Labels:          map[string]string{nfdgc.NRTOwnerPodLabel: fmt.Sprintf("%s/%s", w.kubernetesNamespace, os.Getenv("POD_NAME"))},
```

Comment:
We should also set the label when updating the NRT. Otherwise we get into trouble with pod restarts/updates (when the name changes) because gc doesn't find the pod and GCs the NRT object.